DMCompiler/Compiler/DMPreprocessor/DMPreprocessorParser.cs

```diff
@@ -314,6 +315,38 @@ internal static class DMPreprocessorParser {
 
                     return _defines!.ContainsKey(definedInner.Text) ? 1.0f : 0.0f;
                 }
+                else if (token.Text == "fexists") {
+                    Advance();
+                    if (!Check(TokenType.DM_LeftParenthesis)) {
+                        Error("Expected '(' to begin fexists() expression");
+                        return DegenerateValue;
+                    }
+
+                    Token fexistsInner = Current();
+
+                    if (fexistsInner.Type != TokenType.DM_Resource && fexistsInner.Type != TokenType.DM_ConstantString) {
+                        Error($"Unexpected token {fexistsInner.PrintableText} - resource path expected");
+                        return DegenerateValue;
+                    }
+
+                    Advance();
+                    if (!Check(TokenType.DM_RightParenthesis)) {
+                        DMCompiler.Emit(WarningCode.DefinedMissingParen, token.Location,
+                            "Expected ')' to end fexists() expression");
+                    }
+
+
+                    var filePath = Path.GetRelativePath(".", fexistsInner.Value.ToString().Replace('\\', '/'));
+
+                    var outputDir = Path.GetDirectoryName(DMCompiler.Settings.Files?[0]) ?? "/";
```

Comment:
In BYOND `#if fexists()` is relative to the current file and not the root of the project.